tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -534,52 +534,10 @@ object CodeGen {
     }
 
     def genChars(value: String): Unit = {
-      // `value` should contain a content of a CString literal as is in its source file
+      // `value` should contain a content of a CString literal as is in its source file parsed with parsed escaped characters
       // malformed literals are assumed absent
       str("c\"")
-      @tailrec def loop(from: Int): Unit =
-        value.indexOf('\\', from) match {
-          case -1 => str(value.substring(from))
-          case idx =>
-            str(value.substring(from, idx))
-            import Character.isDigit
-            def isOct(c: Char): Boolean = isDigit(c) && c != '8' && c != '9'
-            def isHex(c: Char): Boolean =
-              isDigit(c) ||
-                c == 'a' || c == 'b' || c == 'c' || c == 'd' || c == 'e' || c == 'f' ||
-                c == 'A' || c == 'B' || c == 'C' || c == 'D' || c == 'E' || c == 'F'
-            value(idx + 1) match {
-              case c @ ('\'' | '"' | '?') => str(c); loop(idx + 2)
-              case '\\'                   => str("\\\\"); loop(idx + 2)
-              case 'a'                    => str("\\07"); loop(idx + 2)
-              case 'b'                    => str("\\08"); loop(idx + 2)
-              case 'f'                    => str("\\0C"); loop(idx + 2)
-              case 'n'                    => str("\\0A"); loop(idx + 2)
-              case 'r'                    => str("\\0D"); loop(idx + 2)
-              case 't'                    => str("\\09"); loop(idx + 2)
-              case 'v'                    => str("\\0B"); loop(idx + 2)
-              case d if isOct(d) =>
-                val oct = value.drop(idx + 1).take(3).takeWhile(isOct)
-                val hex =
-                  Integer.toHexString(Integer.parseInt(oct, 8)).toUpperCase
-                str {
-                  if (hex.length < 2) "\\0" + hex
-                  else "\\" + hex
-                }
-                loop(idx + 1 + oct.length)
-              case 'x' =>
-                val hex = value.drop(idx + 2).takeWhile(isHex).toUpperCase
-                str {
-                  if (hex.length < 2) "\\0" + hex
-                  else "\\" + hex
-                }
-                loop(idx + 2 + hex.length)
-              case unknown =>
-                // clang warns but allows unknown escape sequences, while java emits errors
-                str(unknown); loop(idx + 2)
-            }
-        }
-      loop(0)
+      str(value.replace("\\", "\\" * 2))
```

Comment:
We also have to escape other characters that would be treated specially by LLVM's parser. At the very least, the character `"` must be escaped, but I'm sure other characters need escaping as well. New lines are likely candidates, as well as all control characters. It's unclear from the LLVM docs what the possible escape formats are, however it's clear that double-hex-digit bytes are allowed, so we should use that for all characters that aren't safe.

Reply:
All characters other then digits and letters would not be encoded as double-hex-digit (only `\` is encoded as `\\`). One of the problems I've noticed was that LLVM treats `\"'` as two characters, which means it had to be encoded as `\22` anyway. 
I'm only wondering does encoding all whitespaces as hex values is good solution. It's the safest option  spaces or tabs are currently also hex values. Maybe I should have change that?

Reply:
I suggest to encode:

* Every non-ASCII character (> 0x7F)
* All control characters (< 0x20 and also 0x7F), so that includes tab but not space (which is = 0x20)
* `"` itself, obviously

Concretely, if `c < 0x20 || c == '"' || c >= 0x7f`, then encode it, otherwise don't encode it.

And I agree that we should always encode using the format `\xx` where `xx` is a 2-digit hex value.